### PR TITLE
Add rules for Base.literal_pow

### PR DESCRIPTION
**Pull request by dfdx**
Alternative to #461

### CODE REVIEW

File: src/rulesets/Base/base.jl [hunk 1]
```diff
@@ -161,3 +161,17 @@ end
...
+function frule((_, _, Δx, _), ::typeof(Base.literal_pow), ::typeof(^), x::Real, pv::Val{p}) where p
+    y = Base.literal_pow(^, x, pv)
+    return y, (p * y / x * Δx)
+end
+
+function rrule(::typeof(Base.literal_pow), ::typeof(^), x::Real, pv::Val{p}) where p
+    y = Base.literal_pow(^, x, pv)
+    function literal_pow_pullback(dy)
+        return NoTangent(), NoTangent(), (p * y / x * dy), NoTangent()
+    end
```
Comment (oxinabox) [inline src/rulesets/Base/base.jl:175]: This fits in the line limit
```suggestion
    literal_pow_pullback(dy) = NoTangent(), NoTangent(), (p * y / x * dy), NoTangent()
```

File: test/rulesets/Base/base.jl [hunk 2]
```diff
@@ -176,4 +176,10 @@
         @test frule((NoTangent(), NoTangent(), NoTangent()), Base.depwarn, "message", :f) !== nothing
         @test rrule(Base.depwarn, "message", :f) !== nothing
     end
+
+    @testset "literal_pow" begin
+        # for real x and n, x must be >0
+        test_frule(Base.literal_pow, ^, rand() + 3, Val(3))
+        test_rrule(Base.literal_pow, ^, rand() + 3, Val(3))
```
Comment (oxinabox) [inline test/rulesets/Base/base.jl:183]: I would like to cut down on use of `rand`
```suggestion
        test_frule(Base.literal_pow, ^, 3.5, Val(3))
        test_rrule(Base.literal_pow, ^, 3.5, Val(3))
```

**Comment by oxinabox** [comment]
I like this one better

**Comment by dfdx** [comment]
Great! I also rebased and bumped the version, so there should be no conflicts (running the workflows still requires the approval though).

**Review by oxinabox — APPROVED** [review]

**Pull request merged**